go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy_controller.go

```diff
@@ -50,6 +50,14 @@ type Controller struct {
 	// An address set factory that creates address sets
 	addressSetFactory addressset.AddressSetFactory
 
+	// anp name is key -> cloned value of ANP kapi is value
+	anpCache map[string]*adminNetworkPolicyState
```

Comment:
I am bit surprised on how heavily reliant this controller is on a cache that is so extensive. This is worthy of a discussion given the memory requirements we have to meet.

The first reason I expected a cache for was to avoid having to iterate all the ANPs to find out which of them applied for a given pod/namespace event. I was not even sure if we would want to have this but the cache needed for this would probably be relatively small.

But having a cache so extensive to determine which kind of OPs to send to the DB, when the DB might be able to just short circuit easily by just comparing the data (or maybe we could implement this ourselves against the libovsdb cache in model client) might be just too much cost.

Reply:
ack let's talk next week.. will set something up..(gmeet/huddle)

Reply:
oh I forgot to ping you regarding this comment.. oops..
I acknowledge this comment as well and see your concern.. Ideally we do want to use the libovsdb caches directly, but that wasn't super easy to translate from kapi which is why I went with the current design of caches

```
But having a cache so extensive to determine which kind of OPs to send to the DB, when the DB might be able to just short circuit easily by just comparing the data (or maybe we could implement this ourselves against the libovsdb cache in model client) might be just too much cost in memory footprint.
```
very true specially with the IC model. For the next release I have some cycles slated out to do some scale runs for ANP - are you ok if we proceed with the current cache model for now, but I will leave this comment unresolved and we revisit this once I have some numbers on the extra mem we are adding? Because as of now we plan to support max 100 ANPs, let me talk to scale team and CUs to see a realistic number of namespaces/pods they are using and how much that bloats the MEM per controller.